unity/Assets/Scripts/Unity/Input/MouseEventsHandler.cs

```diff
@@ -54,29 +56,77 @@ void Update()
                     {
                         oldTile = currentCollisionTile;
                         currentCollisionTile = hit.collider.gameObject;
-                        updateTiles();
+                        UpdateTiles();
                     }
                     isMouseOverTile = true;
                 }
                 else
                 {
+                    GameObjectFinder.UnityMap.ResetGameScore();
                     isMouseOverTile = false;
+                    oldTile = currentCollisionTile;
+                    currentCollisionTile = null;
+                    UpdateTiles();
                 }
             }
         }
 
-        private void updateTiles()
+        private void SetHighlightedTiles()
+        {
+            SetHighlightedTilesState(false);
+            UpdateHighlightedTiles();
+            SetHighlightedTilesState(true);
+            SetPossibleScoreChange();
```

Comment:
Doing this twice looks redundant to me. Quick test also seems to work, if only line 78 and 79 are active.

Reply:
It is needed because:
- there are certain events where the highlight wasnt enabled on all tiles
- this had mostly to do with updates getting in between eachother. I'm not entirely sure how unity handles it in the background, but I was having weird issues where entire tilesets where being ignored and therefor kept highlighted